src/api_server/src/lib.rs

```diff
@@ -217,6 +241,14 @@ impl ApiServer {
         response
     }
 
+    #[cfg(target_arch = "x86_64")]
```

Comment:
nit: instead of making the function `x86_64` specific, maybe we can make just it's internal logic specific. The function could then be called on all platforms.

Reply:
Due to Clippy errors, this will be done better when we have more fatal errors to handle, through a larger 'match'.